android/titanium/src/java/org/appcelerator/titanium/view/TiToolbarStyleHandler.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2018 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+package org.appcelerator.titanium.view;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.content.res.TypedArray;
+import android.support.v7.widget.Toolbar;
+import java.lang.reflect.Field;
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.util.TiRHelper;
+
+/**
+ * Updates a toolbar's height and font size when the orientation changes.
+ * Intended to be used by activities which override "configChanges" for orientation since the system won't
+ * automatically update toolbars in that case (56dp for portrait and 48dp for landscape).
+ * <p>
+ * The owner of this handler is expected to call its onConfigurationChanged() method when its activity
+ * equivalent method has been invoked.
+ */
+public class TiToolbarStyleHandler
+{
+	/** The default Android log tag name to be used by this class. */
+	private static final String TAG = "TiToolbarStyleHandler";
+
+	/** The toolbar to be updated with the newest style/size. */
+	private Toolbar toolbar;
+
+	/**
+	 * Creates a new handler for the given toolbar.
+	 * @param toolbar The toolbar to be resized by this handler. Cannot be null.
+	 */
+	public TiToolbarStyleHandler(Toolbar toolbar)
+	{
+		if (toolbar == null) {
+			throw new NullPointerException();
+		}
+
+		this.toolbar = toolbar;
+	}
+
+	/**
+	 * Gets the toolbar that is being handled by this instance.
+	 * @return Returns the toolbar being handled.
+	 */
+	public Toolbar getToolbar()
+	{
+		return this.toolbar;
+	}
+
+	/**
+	 * To be called by the owner when the activity/view's overriden onConfigurationChanged() method has been called.
+	 * Updates the toolbar's height and font size base on the given configuration.
+	 * @param newConfig The udpated configuration applied to the activity/view.
+	 */
+	public void onConfigurationChanged(Configuration newConfig)
```

Comment:
The `newConfig` parameter seems to not be used in the method. It is used when the method is called from:
https://github.com/appcelerator/titanium_mobile/pull/10383/files#diff-9ea98253bfde40055a91a9df7a25cb33R52
and 
https://github.com/appcelerator/titanium_mobile/pull/10383/files#diff-9ce4241f5edf3cb3a0918ec183f9f59bR83
Maybe the former one may allow us to remove the parameter from the definition in `TiActionBarStyleHandler` too?

Reply:
I made it this way to make it clear that this method should be called when the Activity/View object's `onConfigurationChanged()` method has been called. That's also partly why I decided to name the class a "handler" because it's like an event handler.

Also, I was originally thinking about optimizing this by only updating the ActionBar/Toolbar when the given configuration object indicated that the orientation changed. But `onConfigurationChanged()` is actually rarely called. So, I didn't think it was worth it.

Are you getting a compiler warning? I may need to add a `@SuppressWarnings("unused")`.

Reply:
I just saw it was unused. I am OK to leave it as is, matching the declaration makes sense for it being more clear what is happening.